muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/workspace-materials.js

```diff
@@ -83,6 +91,20 @@
           .text(getLocaleText("plugin.workspace.materialsLoader.evaluatedAssignmentLabel"))
       );
     });
+
+    $('.workspace-materials-view-page').waypoint(function(direction) {
+      if ($(window).data('scrolling') !== true && $(window).data('initializing') !== true) {
+        var workspaceMaterialId = $(this).data('workspace-material-id');
+        $('a.active').removeClass('active');
```

Comment:
looking for any links with "active" class is too broad, you should limit to links inside TOC